Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -1356,7 +1353,18 @@ CompletionHandler<void(Decoder*)> takeAsyncReplyHandler(Connection& connection,
 
 void Connection::wakeUpRunLoop()
 {
-    RunLoop::main().wakeUp();
+    if (!isValid())
+        return;
+    m_syncState->runLoop().wakeUp();
```

Comment:
Confusing to be getting a runLoop from m_syncState when this code is not about sync IPC

Reply:
I moved these into a helper that explains it a bit better, and asserts the required conditions.